examples/demo_bgp.pp

```diff
@@ -130,46 +130,72 @@
   #---------------------------------------------------------------------------#
   # Configure BGP IPv4 Neighbors
   #---------------------------------------------------------------------------#
+
+  $capability_negotiation = $operatingsystem ? {
+    'nexus' => true,
+    default => undef
+  }
+  $dynamic_capability = $operatingsystem ? {
+    'nexus' => true,
+    default => undef
+  }
+  $log_neighbor_changes = $operatingsystem ? {
+    'nexus' => disable,
+    default => undef
+  }
+  $low_memory_exempt = $operatingsystem ? {
+    'nexus' => false,
+    default => undef
+  }
+  $remove_private_as = $operatingsystem ? {
+    'nexus' => 'all',
+    default => undef
+  }
+  $update_source = $operatingsystem ? {
+    'nexus' => 'ethernet1/1',
+    default => 'fastethernet1/1/1/1'
+  }
+
   cisco_bgp_neighbor {'55.77 blue 1.1.1.1':
     ensure                 => present,
 
     #Properties
     description            => 'my description',
     connected_check        => true,
-    capability_negotiation => true,
-    dynamic_capability     => true,
+    capability_negotiation => $capability_negotiation,
+    dynamic_capability     => $dynamic_capability,
     ebgp_multihop          => 2,
-    local_as               => 55.77,
-    log_neighbor_changes   => disable,
-    low_memory_exempt      => false,
+    local_as               => 55.88,
```

Comment:
Yes.  XR doesn't allow a neighbor local_as to be the same as the BGP AS, so I just changed it for both XR and NX.